trame_rca/utils.py

```diff
@@ -62,44 +73,98 @@ def encode(
         return self._impl(np_image, self.value, cols, rows, quality, now_ms)
 
 
-class VtkImageExtract:
-    def __init__(self, render_window):
-        self.render_window = render_window
-        self.window_to_image = vtkWindowToImageFilter()
-        self.window_to_image.SetInput(render_window)
-        self.window_to_image.SetScale(1)
-        self.window_to_image.ReadFrontBufferOff()
-        self.window_to_image.ShouldRerenderOff()
-        self.window_to_image.FixBoundaryOn()
+class AbstractWindow(ABC):
+    """
+    Abstract base class that wraps a remote window to interacts with the RCA.
+
+    Subclasses of `AbstractWindow` must implement the defined abstract methods
+    """
+
+    @property
+    @abstractmethod
+    def img_cols_rows(self) -> tuple[NDArray, int, int]:
+        """
+        Returns a tuple containing the window as a numpy array,
+        the number of cols and the number of rows.
+        This method is called by the scheduler to render the window.
+        """
+        pass
+
+    @abstractmethod
+    def process_resize_event(self, width: int, height: int) -> None:
+        """
+        Make the window process any resizing event on the RCA.
+        This method is called by the adapter when a resizing event happens.
+        """
+        pass
+
+    @abstractmethod
+    def process_interaction_event(self, event: dict) -> None:
+        """
+        Make the window process any interaction event on the RCA.
+        This method is called by the adapter when an interaction event happens.
```

Comment:
Maybe add a link to the list of possible event types ? [is it this list?](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/RenderWindowInteractor/index.js).